src/services/contracts/vault.service.ts

```diff
@@ -35,8 +35,8 @@ export default class VaultService {
     tokenOutAmount: string,
     options: Record<string, any> = {}
   ): Promise<TransactionResponse> {
-    const storeState = store.state as any;
-    const deadline = calculateValidTo(storeState.app.transactionDeadline);
+    const { transactionDeadline } = useAppStore();
```

Comment:
Not a big fan of using composables within services like this. Really `transactionDeadline` should be passed into the service as a param wherever the service is used in the component context.

Reply:
Super agree, refactored.